src/Microsoft.AspNet.Razor/Chunks/Generators/TagHelperDirectiveChunkGenerator.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+
+namespace Microsoft.AspNet.Razor.Chunks.Generators
+{
+    public abstract class TagHelperDirectiveChunkGenerator : SpanChunkGenerator
```

Comment:
Instead of having this common class we could duplicate the `GetDirectiveText` on each of the inheriting `ChunkGenerator`s but it'd still require an interface to avoid ugliness in `TagHelperDirectiveSpanVisitor`. I felt like this was the cleanest approach.